addons/web/static/src/views/fields/properties/properties_field.js

```diff
@@ -63,23 +63,49 @@ export class PropertiesField extends Component {
         this.definitionRecordField = field.definition_record;
 
         this.state = useState({
-            canChangeDefinition: true,
+            canChangeDefinition: false,
+            isEditMode: false,
             movedPropertyName: null,
-            showAddButton: this.props.showAddButton,
             unfoldedSeparators: this._getUnfoldedSeparators(),
         });
 
         // Properties can be added from the cogmenu of the form controller
         if (this.env.config?.viewType === "form") {
-            useBus(this.env.model.bus, "PROPERTY_FIELD:ADD_PROPERTY_VALUE", () => {
-                this.onPropertyCreate();
+            useBus(this.env.model.bus, "PROPERTY_FIELD:EDIT", () => {
+                if (!this.props.readonly && this.state.canChangeDefinition) {
+                    this.state.isEditMode = true;
+                    if (this.propertiesList.length === 0) {
+                        this.onPropertyCreate();
+                    }
+                }
             });
         }
 
         onWillStart(async () => {
-            await this._checkDefinitionAccess();
+            const canChangeDefinition = await this.checkDefinitionWriteAccess();
+            this.state.canChangeDefinition = !!canChangeDefinition;
+        });
+
+        /** @param {function} callback */
+        const whenRecordStoringPropertyDefinitionChanges = (callback) => {
+            const { name } = this.props.record.fields[this.definitionRecordField];
+            useEffect(callback, () => [this.props.record.data[name]]);
+        };
+
+        whenRecordStoringPropertyDefinitionChanges(() => {
+            this.checkDefinitionWriteAccess().then((canChangeDefinition) => {
+                this.state.canChangeDefinition = !!canChangeDefinition;
+            });
         });
 
+        useEffect(
+            (canChangeDefinition, readonly, editMode) => {
+                this.state.isEditMode =
+                    canChangeDefinition && !readonly && (this.state.isEditMode || editMode);
```

Comment:
Does not need to be on its own line.

Reply:
If I remove this extra line, my linter is not happy. I use the tools and configuration set up in `addons/web/tooling/enable.sh`.